movement/watch_faces/complication/randonaut_face.c

```diff
@@ -0,0 +1,411 @@
+
+    // return true if the watch can enter standby mode. Generally speaking, you should always return true.
+    // Exceptions:
+    //  * If you are displaying a color using the low-level watch_set_led_color function, you should return false.
+    //  * If you are sounding the buzzer using the low-level watch_set_buzzer_on function, you should return false.
+    // Note that if you are driving the LED or buzzer using Movement functions like movement_illuminate_led or
+    // movement_play_alarm, you can still return true. This guidance only applies to the low-level watch_ functions.
+    return true;
+}
+
+void randonaut_face_resign(movement_settings_t *settings, void *context) {
+    (void) settings;
+    (void) context;
+
+    // handle any cleanup before your watch face goes off-screen.
+}
+
+// PRIVATE STATIC FUNCTIONS ///////////////////////////////////////////////////
+
+/** @brief display handler
+ */
+static void _randonaut_face_display(randonaut_state_t *state) {
+    char buf[12];
+    watch_clear_colon();
+    switch ( state->face.mode ) {
+        case 0: //home
+            sprintf(buf, "RA   Rando");
+            break;
+        case 1: //generate
+            if ( state->quantum )
+                // All Hail Steve /;[;[/.;]/[.;[/;/;/;/;.;.];.]]--=/ 
+                for ( uint8_t c = 100; c > 0; c--) {////////////////
+                    ___0xf322(__0x2_(/*0xD68 _0x22*//*__0x22*///////
+                    /*_0x22*/0x2),__0x2_(/*  _0x22*//*_0x22*////////
+                    0x33-0x1C));___0xf322(__0x2_(/*___0x2222_22___*/
+                    /*0x2*/0x2),__0x2_(3432/*_0x22*//*_0x2222*//////
+                    -3409));___0xf322(__0x2_(/*0x2*//*____0x222222*/
+                    002),__0x2_(0xE +9));___0xf322(/*_0x2222222222*/
+                    __0x2_(0x2),__0x2_(23));___0xf322(/*____0x2222*/
+                    /*0x2*/__0x2_(002),__0x2_(12+7+11));/*___00x22*/
+                    if(/*_0x22*/c<70){_0x22(__0x2_(2),/*____0x2222*/
+                    __0x2_(12+7+11));}if(c<60){_0x22(/*_______0x22*/
+                    /*_0x22*/__0x2_(002),__0x2_(0xD68-/*_0x2222222*/
+                    0xD4A));}if(c<50){_0x22(__0x2_(0x2),/*____0x22*/
+                    __0x2_(14+9));}delay_ms(__0x2_(c)+20);if/*_0x2*/
+                    (c<30){watch_display_string(" ",__0x2_(/*_2**2*/
+                    10));}_0x22(__0x2_(02),__0x2_(3432-3409)/*0x22*/
+                    );_0x22(__0x2_(002),__0x2_(51-28));/*_____0x22*/
+                    /**/  _0x22(__0x2_(0x2),__0x2_(23));if(c<20)////
+                    /*_*/{_0x22(__0x2_(02),__0x2_(51-28));/*__0x22*/
+                    /*_0x22*/_0x22(__0x2_(2),__0x2_(14+9));/*_0x22*/
+                    /*_0x22*/_0x22(__0x2_(0x2),__0x2_(0xD68-0xD4A));
+                    /*_0x22*/_0x22(__0x2_(0x2),__0x2_(3432-3409));//
+                    /*_0x22*/_0x22(__0x2_(002),__0x2_(12+7+11));////
+                    /*_0x22**_0x22*/_0x22(__0x2_(2),__0x2_(51-28));}
+                }
```

Comment:
What on earth is going on here? Code like this makes me deeply uncomfortable.

Reply:
oh lol itś actually just an animation randomly turning pixels on and off. the obfuscation was merely for artistic purposes.

itś really just these three function pointers (lines 50-52) and numeric values written in a nondecimal way...

```C
static uint32_t (*__0x2_)(uint32_t) = &_get_pseudo_entropy;
static void (*_0x22)(uint8_t,uint8_t) = &watch_clear_pixel;
static void (*___0xf322)(uint8_t,uint8_t) = &watch_set_pixel;
```

but if it makes you uncomfortable I get it. I can change it back to human readable code if you want.